src/wilcoxon.jl

```diff
@@ -163,6 +163,27 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
     end
 end
 
+function psignrank(W, n, lower_tail)
+    int_type = typeof(n)
+    W = int_type(W)
+    two = one(int_type) + one(int_type)
+    max_W =  n * (n + one(int_type)) ÷ two + one(int_type)
+    DP = zeros(int_type(n), max_W)
+    DP[begin] = one(int_type)
+    for j in one(int_type):n
+        for i in max_W:-one(int_type):one(int_type)
+            if i - j >= one(int_type)
+                DP[i] += DP[i - j]
+            end
+        end
+    end
+    if lower_tail
+        return exp(log(sum(@view DP[begin:W+one(int_type)])) - n*log(two)) # exp log to avoid 2^n
+    else
+        return exp(log(sum(@view DP[W+one(int_type):end])) - n*log(two))
```

Comment:
```suggestion
        return ldexp(sum(@view DP[(W + 1):end]), -n)
```

Reply:
I am not familiar with `ldexp`.

Reply:
`ldexp(x, n)` returns `x * 2^n`: https://docs.julialang.org/en/v1/base/math/#Base.Math.ldexp

Reply:
Ok, that seems like the calculation we need.
However:
```jl
julia> ldexp(5,2)
ERROR: MethodError: no method matching ldexp(::Int64, ::Int64)

julia> ldexp(5.0,2)
20.0  
```

Reply:
We could use
```suggestion
return ldexp(float(sum(@view DP[(W + 1):end])), -n)
```